src/decision/location_selection.jl

```diff
@@ -0,0 +1,266 @@
+    target_seed_sites=nothing, 
+    target_shade_sites=nothing)
+
+    ranks = run_site_selection(domain, 
+        scenarios, 
+        sum_cover, 
+        area_to_seed; 
+        target_seed_sites=target_seed_sites,
+        target_shade_sites=target_shade_sites,
+    )
+
+    return aggregation_function(ranks(; intervention=iv_type))
+
+end
+
+
+"""
+    ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)ranks_to_frequencies(rs::ResultSet, iv_type::String; n_ranks=length(ranks.sites))
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,3,A};n_ranks=length(ranks.sites),agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),) where {D,T,A}
+    ranks_to_frequencies(ranks::NamedDimsArray{D,T,2,A};n_ranks=length(ranks.sites),agg_func=nothing) where {D,T,A}
+
+Post-processing function for location ranks output of `run_location_selection()`. Gives the frequency 
+with which each location was selected at each rank across the location selection scenarios.
+
+# Arguments
+- `ranks` : Contains location ranks for each scenario of location selection, as created by 
+    `run_location_selection()`.
+- `n_ranks` : number of rankings (defualt is number of locations).
+- `agg_func` : Aggregation function to appy after frequencies are calculated.
+
+# Returns
+Frequency with which each location was selected for each rank.
+"""
+function ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)
+    dn = NamedDims.dimnames(ranks)
+    freq_dims = [n for n in dn if n != :scenarios]
+    dn_subset = vcat(freq_dims, [:ranks])
+    freq_elements = vcat(
+        [1:size(ranks, n) for n in dn if n != :scenarios],
+        [1:size(ranks, :sites)],
+    )
+    mn = ([size(ranks, k) for k in freq_dims]..., size(ranks, :sites))
+
+    rank_frequencies = NamedDimsArray(zeros(mn...); zip(dn_subset, freq_elements)...)
+
+    for rank in 1:n_ranks
+        rank_frequencies[ranks=Int64(rank)] .= sum(ranks .== rank; dims=:scenarios)[scenarios=1]
+    end
+
+    return rank_frequencies
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,3,A};
+    n_ranks=length(ranks.sites),
+    agg_func=x -> dropdims(sum(x; dims=:timesteps); dims=:timesteps),
+) where {D,T,A}
+    return agg_func(ranks_to_frequencies(ranks, n_ranks))
+end
+function ranks_to_frequencies(
+    ranks::NamedDimsArray{D,T,2,A};
+    n_ranks=length(ranks.sites),
+    agg_func=nothing,
+) where {D,T,A}
+    if !isnothing(agg_func)
+        return agg_func(ranks_to_frequencies(ranks, n_ranks))
+    end
+
+    return ranks_to_frequencies(ranks, n_ranks)
+end
+
+"""
+    location_selection_frequencies(ranks::NamedDimsArray;n_loc_int::Int64=5)
+    location_selection_frequencies(inv_log::NamedDimsArray{D,T,4,A};dims::Union{Symbol,Vector{Symbol}}=:coral_id) where {D,T,A}
+
+Post-processing function for intervention logs. Calculates the frequencies with which locations were selected for a particular intervention,
+for a selection of scenarios (e.g. selected robust scenarios).
```

Comment:
Good docs state clearly what the function does.
There's no need to say "such and such function", "This function", etc.

We know it's a function - it's why it has documentation.

```
Post-process intervention logs to calculate the frequency with which locations were selected for a particular intervention, for a set of scenarios.
```